grid-client/deployer/deployer.go

```diff
@@ -155,28 +156,49 @@ func (d *Deployer) deploy(
 			log.Debug().Uint32("Number of public ips", publicIPCount)
 
 			contractID, err := d.substrateConn.CreateNodeContract(d.identity, node, dl.Metadata, hashHex, publicIPCount, newDeploymentSolutionProvider[node])
-			log.Debug().Uint64("CreateNodeContract returned id", contractID)
 			if err != nil {
-				return currentDeployments, errors.Wrapf(err, "failed to create contract on node %d", node)
+				if strings.Contains(err.Error(), "ContractIsNotUnique") {
+					contractID, err = d.substrateConn.GetContractWithHash(d.identity, node, hash)
+					if err != nil {
+						return currentDeployments, errors.Wrapf(err, "failed to find existing contract on node %d", node)
+					}
+					log.Info().
+						Uint32("node", node).
+						Uint64("contractID", contractID).
+						Msg("reusing existing contract")
+				} else {
+					return currentDeployments, errors.Wrapf(err, "failed to create contract on node %d", node)
+				}
 			}
-
+			log.Debug().Uint64("returned contract ID", contractID)
 			dl.ContractID = contractID
+
+			// Update deployment with contract ID and send to node
 			err = client.DeploymentDeploy(ctx, dl)
 			if err != nil {
-				rerr := d.substrateConn.EnsureContractCanceled(d.identity, contractID)
-				if rerr != nil {
-					return currentDeployments, errors.Wrapf(err, "error cancelling contract: %s; you must cancel it manually (id: %d)", rerr, contractID)
+				// If deployment exists, continue as already deployed
+				if !errors.Is(err, provision.ErrDeploymentExists) {
```

Comment:
I don't think the errors will match since the first error is received from node client as rmb message not the same reference as provision.ErrDeploymentExists